File: MYCM/support/domain/ddd/Repository.cs
using support.domain.ddd;
using support.persistence.repositories;
using System;
namespace support.domain.ddd{
    /// <summary>
    /// Generic Interface for marking up repositories of domain aggregate roots
    /// Only Aggregate Roots can have a domain entity repository since 
    /// they are the ones who aggregate other entities
    /// </summary>
    /// <typeparam name="E">Generic-Type of the aggregate root which repository is held</typeparam>
    /// <typeparam name="ID">Generic-Type of the persistence entitiy identifier</typeparam>
    /// <typeparam name="EID">Generic-Type of the domain identifier</typeparam>
    //TODO: We have a problem on generic identifiers
    //C# doesn't have an "anonymous generic tag" so we can't do stuff like AggregateRoot<?>
    public interface Repository<E,ID,EID>:DataRepository<E,ID> where E:AggregateRoot<EID>{
        /// <summary>
        /// Finds an entity based on its identitifer
        /// </summary>
        /// <param name="entityID">EID with the entity identifier</param>
        /// <returns>E with the entity which is represend by a certain entity identifier</returns>
        E find(EID entityID);
    }
}

File: MYCA/myca/support/src/main/java/cdiomyc/support/domain/ddd/Repository.java
package cdiomyc.support.domain.ddd;

import cdiomyc.support.persistence.repositories.DataRepository;

/**
 * Generic Interface for marking aggregate root repositories
 * @author <a href="https://github.com/freitzzz">freitzzz</a>
 * @param <E> Generic-Type of the repository aggregate root
 * @param <ID> Generic-Type of the repository aggregate root persistence identifier
 * @param <EID> Generic-Type of the repository aggregate root identifier
 */
public interface Repository<E extends AggregateRoot<EID>,ID,EID> extends DataRepository<E,ID>{
    /**
     * Finds a domain entity by its identifier
     * @param domainEntityIdentifier EID with the domain entity identifier
     * @return E with the domain entity which is identified by a certain identifier
     */
    E findEID(EID domainEntityIdentifier);
}